inc/cropper/namespace.php

```diff
@@ -884,143 +896,46 @@ function massage_meta_data_for_orientation( array $meta_data ) {
 }
 
 /**
- * Filters 'img' elements in post content to add 'srcset' and 'sizes' attributes.
- *
- * @param string $content The raw post content to be filtered.
- * @param string $context The current filter context.
+ * Add our special handlers for width & height attrs and srcset attributes.
  *
- * @return string Converted content with 'srcset' and 'sizes' attributes added to images.
+ * @param string $filtered_image Full img tag with attributes that will replace the source img tag.
+ * @param string $context Additional context, like the current filter name or the function name from where this was called.
+ * @param int $attachment_id  The image attachment ID. May be 0 in case the image is not an attachment.
+ * @return string Full img tag with attributes that will replace the source img tag.
  */
-function filter_content_tags( string $content, string $context = null ) : string {
-	if ( null === $context ) {
-		$context = current_filter();
+function content_img_tag( string $filtered_image, string $context, int $attachment_id ) : string {
+	if ( strpos( $filtered_image, TACHYON_URL ) === false ) {
+		return $filtered_image;
 	}
 
-	if ( false === strpos( $content, '<img' ) ) {
-		return $content;
-	}
+	$image_meta = wp_get_attachment_metadata( $attachment_id );
 
-	$images = Tachyon::parse_images_from_html( $content );
-	if ( empty( $images ) ) {
-		// No images, leave early.
-		return $content;
+	// Add 'width' and 'height' attributes if applicable.
+	if ( $attachment_id > 0 && ! str_contains( $filtered_image, ' width=' ) && ! str_contains( $filtered_image, ' height=' ) ) {
```

Comment:
after following the advice above, we can remove the first check, simplifying the code.

Reply:
core copy/pasta here, you're right it could simplified a bit.